From f456420bd5158edd015c1303b29aaf67253f44fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julius=20R=C3=BCberg?=
 <22551563+Toorero@users.noreply.github.com>
Date: Thu, 19 Aug 2021 23:19:41 +0200
Subject: [PATCH 1/4] Changed to proper generic python command

---

Obtained from: https://github.com/GPUOpen-LibrariesAndSDKs/RadeonProRenderBlenderAddon/pull/345
Minor patch changes by Orson Teodoro <orsonteodoro@hotmail.com>
Document changes for Apache-2.0 compliance.

License header for
BlenderPkg/build_linux.sh:
build.py:
src/bindings/pyrpr/rpr.py:

#**********************************************************************
# Copyright 2020 Advanced Micro Devices, Inc
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#********************************************************************

diff -urp RadeonProRenderBlenderAddon-3.3.16.orig/BlenderPkg/build_linux.sh RadeonProRenderBlenderAddon-3.3.16/BlenderPkg/build_linux.sh
--- RadeonProRenderBlenderAddon-3.3.16.orig/BlenderPkg/build_linux.sh	2021-12-16 11:53:07.000000000 -0800
+++ RadeonProRenderBlenderAddon-3.3.16/BlenderPkg/build_linux.sh	2021-12-18 09:48:42.341570948 -0800
@@ -15,8 +15,10 @@
 # limitations under the License.
 #********************************************************************
 
+# oteodoro: Changed for generic python3 call for systems without python3.7.
+
 pushd ..
 ./build.sh
 popd
 
-python3.7 create_zip_addon.py
+python3 create_zip_addon.py # oteodoro: python3.7 -> python3
diff -urp RadeonProRenderBlenderAddon-3.3.16.orig/build.py RadeonProRenderBlenderAddon-3.3.16/build.py
--- RadeonProRenderBlenderAddon-3.3.16.orig/build.py	2021-12-16 11:53:07.000000000 -0800
+++ RadeonProRenderBlenderAddon-3.3.16/build.py	2021-12-18 09:45:49.241570967 -0800
@@ -41,14 +41,12 @@ subprocess.check_call([sys.executable, '
 subprocess.check_call([sys.executable, 'rpr_load_store.py'])
 os.chdir(cwd)
 
-if sys.version_info.major == 3 and sys.version_info.minor == 9:
-    # we are going to build RPRBlenderHelper only for python 3.9
-    os.chdir('RPRBlenderHelper')
-    shutil.rmtree('.build', ignore_errors=True)
-    os.makedirs('.build')
-    os.chdir('.build')
-    if 'Windows' == platform.system():
-        subprocess.check_call(['cmake', '-G', 'Visual Studio 14 2015 Win64',  '..'])
-    else:
-        subprocess.check_call(['cmake', '..'])
-    subprocess.check_call(['cmake', '--build',  '.', '--config', 'Release', '--clean-first'])
+# Build RPRBlenderHelper									# <changed>
+os.chdir('RPRBlenderHelper')									#
+os.makedirs('.build', exist_ok=True)								#
+os.chdir('.build')										#
+if 'Windows' == platform.system():								# Toorero: python3 minor agnostic
+    subprocess.check_call(['cmake', '-G', 'Visual Studio 14 2015 Win64',  '..'])		#
+else:												#
+    subprocess.check_call(['cmake', '..'])							#
+subprocess.check_call(['cmake', '--build',  '.', '--config', 'Release', '--clean-first'])	# </changed>
Only in RadeonProRenderBlenderAddon-3.3.16: build.py.orig
Only in RadeonProRenderBlenderAddon-3.3.16: build.py.rej
diff -urp RadeonProRenderBlenderAddon-3.3.16.orig/build.sh RadeonProRenderBlenderAddon-3.3.16/build.sh
--- RadeonProRenderBlenderAddon-3.3.16.orig/build.sh	2021-12-16 11:53:07.000000000 -0800
+++ RadeonProRenderBlenderAddon-3.3.16/build.sh	2021-12-18 09:44:38.537710291 -0800
@@ -1,10 +1,10 @@
 cxml="/usr/bin/castxml"
 if [ -f "$cxml" ]; then
-    python3.7 cmd_tools/create_sdk.py
-	python3.7 src/bindings/pyrpr/src/pyrprapi.py $cxml
+    python3 cmd_tools/create_sdk.py								# oteodoro: python3 minor agnostic
+	python3 src/bindings/pyrpr/src/pyrprapi.py $cxml					# oteodoro:
 	if [ -f "./bindings-ok" ]; then
-		python3.7 build.py
-		python3.9 build.py
+		python3 build.py								# oteodoro: python3 minor agnostic
+		python3 build.py								# oteodoro
 	else
 		echo Compiling bindings failed
 	fi
diff -urp RadeonProRenderBlenderAddon-3.3.16.orig/src/bindings/pyrpr/rpr.py RadeonProRenderBlenderAddon-3.3.16/src/bindings/pyrpr/rpr.py
--- RadeonProRenderBlenderAddon-3.3.16.orig/src/bindings/pyrpr/rpr.py	2021-12-18 09:43:39.211309265 -0800
+++ RadeonProRenderBlenderAddon-3.3.16/src/bindings/pyrpr/rpr.py	2021-12-18 09:47:53.155582435 -0800
@@ -97,7 +97,7 @@ def export(json_file_name, dependencies,
         if distro.id() != 'gentoo':                                                                     # oteodoro added line
           cffi_libs_dir = Path(_cffi_backend.__file__).parent / '.libs_cffi_backend'                    # oteodoro changed indent
           if not cffi_libs_dir.is_dir():                                                                # oteodoro changed indent
-              cffi_libs_dir = Path(_cffi_backend.__file__).parent / 'cffi.libs'                         # oteodoro changed indent
+              cffi_libs_dir = Path(_cffi_backend.__file__).parent / 'cffi'                              # oteodoro changed indent, Toorero changed
 
           for path in cffi_libs_dir.iterdir():                                                          # oteodoro changed indent
               if '.so' in path.suffixes:                                                                # oteodoro changed indent
Only in RadeonProRenderBlenderAddon-3.3.16/src/bindings/pyrpr: rpr.py.orig
Only in RadeonProRenderBlenderAddon-3.3.16/src/bindings/pyrpr: rpr.py.rej
